source/java/src/org/lucee/extension/cache/eh/util/EHCacheClassLoader.java

```diff
@@ -0,0 +1,135 @@
+package org.lucee.extension.cache.eh.util;
+
+import java.lang.reflect.Method;
+
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+
+import lucee.loader.engine.CFMLEngine;
+import lucee.loader.engine.CFMLEngineFactory;
+import lucee.runtime.config.Config;
+
+import lucee.commons.io.log.Log;
+
+
+/**
+ * A special class loader that first attempts to load class from the Lucee Environment,
+ * but that can fail when RMI classes are requested via a spawned thread, so when
+ * the class cannot be located in the Lucee Environment, it searches all the Felix
+ * bundles for the requested class. This will prevent issues with issues trying
+ * to load classes like net.sf.ehcache.distribution.RMICachePeer_Stub. 
+ *
+ * @author Dan G. Switzer, II
+ * @version 1.0
+ */
+public class EHCacheClassLoader extends ClassLoader {
+	private Config config;
+	private ClassLoader LuceeEnvClassLoader;
+
+	public EHCacheClassLoader(Config config) {
+		this.config = config;
+
+		try {
+			Method m = config.getClass().getMethod("getClassLoaderEnv", new Class[0]);
+			this.LuceeEnvClassLoader = (ClassLoader) m.invoke(config, new Object[0]);
+		}
+		catch (Exception e) {
+			// do nothing
+		}
+	}
+
+	public CFMLEngine getCFMLEngine(){
+		try {
+			Class<?> clazz = CFMLEngineFactory.getInstance().getClassUtil().loadClass("lucee.runtime.config.ConfigWebUtil");
+			Method m = clazz.getMethod("getEngine", new Class[] { Config.class });
+			return (CFMLEngine) m.invoke(null, this.config);
+		} catch (Exception e) {
+			e.printStackTrace();
+		}
+
+		return null;
+	}
+
+	public boolean isFrameworkBundle(Bundle b){
+		try {
```

Comment:
we need to mark this in the Lucee core, that this is used in that extension and it should not be changed.

Reply:
Is there some action I should be taking in this pull request?

Reply:
please just comment direct referecnes to core methods like this in the descriptipn of the PR

Reply:
@michaeloffner,

I've added a comment to [EHCacheClassLoader.java](https://github.com/lucee/extension-ehcache/pull/1/files/4d2697dcedcf137712aa6ccebaa56132b40ab17d#diff-75fb84c3ef41b825e246fb4952b953aae3252653bb22918e635aef25cd7fa842) and also updated the PR description above.